src/TweetProcessor.js

```diff
@@ -277,11 +277,21 @@ class TweetProcessor {
       promise = Promise.resolve();
     }
 
-    promise = promise.then(() => { return this._twitter.profile.update(tweet.user); });
+    // If all the rooms are marked as hot, don't bother with updating the profile.
+    const updateProfiles = ![...rooms].every((roomId) => {return opts.hotRooms.has(roomId)});
```

Comment:
Potentially, but it's a fairly cheap thing to check. The case is usually where you have one or two hot rooms such as a popular hashtag room where we are unlikely to ever use the profile and it would be a bit silly to store it.